src/framework/global/types/translatablestring.h

```diff
@@ -0,0 +1,173 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2021 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+#ifndef MU_FRAMEWORK_TRANSLATABLESTRING_H
+#define MU_FRAMEWORK_TRANSLATABLESTRING_H
+
+#include "translation.h"
+
+#include "types/string.h"
+
+#ifndef NO_QT_SUPPORT
+#include <QString>
+#endif
+
+namespace mu {
+//! Note: in order to make the string visible for `lupdate`,
+//! you must write TranslatableString(...) explicitly.
+struct TranslatableString {
+    const char* context = nullptr;
+    const char* str = nullptr;
+    const char* disambiguation = nullptr;
+
+    TranslatableString() = default;
+    TranslatableString(const char* context, const char* str, const char* disambiguation = nullptr)
+        : context(context), str(str), disambiguation(disambiguation) {}
```

Comment:
I think the constructor will need an `int n` parameter so that `lrelease` knows to request translations with plurals (look for `<numerusform>` in the .ts files). You could store `n` as a member variable but still allow requesting translations with a different number later on:

```c++
    inline String translated(int num = -1) const
    {
        // ...

        assert(num >= -1);
        assert(num == -1 || n != -1);
        String res = mtrc(context, str, disambiguation, num == -1 ? n : num);

        // ...

        return res;
    }
```
With `assert(n >= -1)` in the constructor.

Reply:
Good point. I believe `lupdate` won't look at the signature of this constructor though, only at the callers. But indeed, if you pass a string containing `%n`, but don't specify a fourth parameter, `lupdate` is apparently confused and gives a warning. 
Currently, we actually never need to use TranslatableString with %n strings, but when we start doing that, we should keep this in mind.